Windows tests in main are failing

### Component(s)

_No response_

### Describe the issue you're reporting

Windows tests in main are failing: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7197037922/job/19603245185

```
package github.com/open-telemetry/opentelemetry-collector-contrib/receiver/namedpipereceiver
	imports github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/namedpipe: build constraints exclude all Go files in D:\a\opentelemetry-collector-contrib\opentelemetry-collector-contrib\pkg\stanza\operator\input\namedpipe
make[2]: *** [../../Makefile.Common:111: test] Error 1
make[2]: Leaving directory 'D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/namedpipereceiver'
make[1]: *** [Makefile:152: receiver/namedpipereceiver] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/pkg/stanza/operator/input/namedpipe/config.go
+++ b/pkg/stanza/operator/input/namedpipe/config.go
@@ -1,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package namedpipe // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/namedpipe"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/split"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/trim"
+)
+
+const (
+	operatorType      = "namedpipe"
+	DefaultMaxLogSize = 1024 * 1024
+)
+
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfig creates a new stdin input config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorID, operatorType),
+	}
+}
+
+// Config is the configuration of a stdin input operator.
+type Config struct {
+	helper.InputConfig `mapstructure:",squash"`
+	BaseConfig         `mapstructure:",squash"`
+}
+
+type BaseConfig struct {
+	Path        string       `mapstructure:"path"`
+	Permissions uint32       `mapstructure:"mode"`
+	Encoding    string       `mapstructure:"encoding"`
+	SplitConfig split.Config `mapstructure:"multiline,omitempty"`
+	TrimConfig  trim.Config  `mapstructure:",squash"`
+	MaxLogSize  int          `mapstructure:"max_log_size"`
+}

--- a/pkg/stanza/operator/input/namedpipe/namedpipe.go
+++ b/pkg/stanza/operator/input/namedpipe/namedpipe.go
@@ -19,43 +19,11 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/decode"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/split"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/trim"
-)
-
-const (
-	operatorType      = "namedpipe"
-	DefaultMaxLogSize = 1024 * 1024
 )
 
 func init() {
 	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
-}
-
-func NewConfig() *Config {
-	return NewConfigWithID(operatorType)
-}
-
-// NewConfig creates a new stdin input config with default values
-func NewConfigWithID(operatorID string) *Config {
-	return &Config{
-		InputConfig: helper.NewInputConfig(operatorID, operatorType),
-	}
-}
-
-// Config is the configuration of a stdin input operator.
-type Config struct {
-	helper.InputConfig `mapstructure:",squash"`
-	BaseConfig         `mapstructure:",squash"`
-}
-
-type BaseConfig struct {
-	Path        string       `mapstructure:"path"`
-	Permissions uint32       `mapstructure:"mode"`
-	Encoding    string       `mapstructure:"encoding"`
-	SplitConfig split.Config `mapstructure:"multiline,omitempty"`
-	TrimConfig  trim.Config  `mapstructure:",squash"`
-	MaxLogSize  int          `mapstructure:"max_log_size"`
 }
 
 // Build will build a namedpipe input operator.

--- a/pkg/stanza/operator/input/namedpipe/namedpipe_nonlinux.go
+++ b/pkg/stanza/operator/input/namedpipe/namedpipe_nonlinux.go
@@ -1,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !linux
+// +build !linux
+
+package namedpipe // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/namedpipe"
+
+import (
+	"errors"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"go.uber.org/zap"
+)
+
+func (c *Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
+	return nil, errors.New("namedpipe input operator is only supported on linux")
+}

--- a/receiver/namedpipereceiver/namedpipe_test.go
+++ b/receiver/namedpipereceiver/namedpipe_test.go
@@ -1,6 +1,9 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+//go:build linux
+// +build linux
+
 package namedpipereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/namedpipereceiver"
 
 import (

